src/main/java/uk/gov/companieshouse/insolvency/data/service/InsolvencyService.java

```diff
@@ -1,46 +1,16 @@
 package uk.gov.companieshouse.insolvency.data.service;
 
-import org.springframework.stereotype.Service;
-import org.springframework.transaction.annotation.Transactional;
-import uk.gov.companieshouse.api.insolvency.CompanyInsolvency;
 import uk.gov.companieshouse.api.insolvency.InternalCompanyInsolvency;
-import uk.gov.companieshouse.api.insolvency.InternalData;
-import uk.gov.companieshouse.insolvency.data.model.InsolvencyDocument;
-import uk.gov.companieshouse.insolvency.data.model.Updated;
-import uk.gov.companieshouse.insolvency.data.repository.InsolvencyRepository;
 
-@Service
-public class InsolvencyService {
-
-    private final InsolvencyRepository insolvencyRepository;
-
-    /**
-     * Insolvency service to store insolvency data onto mongodb and call chs kafka endpoint.
-     * @param insolvencyRepository repository to save data to db
-     */
-    public InsolvencyService(InsolvencyRepository insolvencyRepository) {
-        this.insolvencyRepository = insolvencyRepository;
-
-    }
+public interface InsolvencyService {
 
     /**
-     * Persist company insolvency information to mongodb collection.
-     * @param insolvencyApi company insolvency information {@link InternalCompanyInsolvency}
+     * Processing company insolvency information.
+     * 1. Saving company insolvency to mongodb collection
+     * 2. call chs-kafka api endpoint
+     * @param companyNumber company number
+     * @param companyInsolvency company insolvency information {@link InternalCompanyInsolvency}
      */
-    public void saveInsolvency(String companyNumber, InternalCompanyInsolvency insolvencyApi) {
-        InsolvencyDocument insolvencyDocument = mapInsolvencyDocument(companyNumber, insolvencyApi);
-
-        insolvencyRepository.save(insolvencyDocument);
-    }
-
-    private InsolvencyDocument mapInsolvencyDocument(String companyNumber,
```

Comment:
Should this interface have `mapInsolvencyDocument` - both classes implementing it have that method

Reply:
Simple method which has got 2 lines of data mapping, we can move it separate class when we got common code in other places. May be like util class.